File: examples/BoundsBug2.java
public class BoundsBug2 implements java.io.Serializable {
  protected Boolean _var14;
  public BoundsBug2() {
    clear();
  }

  public BoundsBug2(Integer executions, Boolean bugHappened) {
    _var14 = bugHappened;
  }
  public void clear() {
    _var14 = false;
  }
  public Boolean didBugHappen() {
    return _var14;
  }
  public void exec() {
  }
}

File: examples/boundsbug2.cpp
#pragma once
#include <algorithm>
#include <functional>
#include <vector>
#include <unordered_set>
#include <string>
#include <unordered_map>

class BoundsBug2 {
public:
protected:
  bool _var14 : 1;
public:
  inline BoundsBug2() {
    _var14 = false;
  }
  explicit inline BoundsBug2(int executions, bool bugHappened) {
    _var14 = bugHappened;
  }
  BoundsBug2(const BoundsBug2& other) = delete;
  inline bool  didBugHappen() {
    return _var14;
  }
  inline void exec () {
  }
};
